p2p/src/peer_manager/peers_eviction.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::BTreeMap, hash::Hasher};
+
+use crypto::random::Rng;
+
+use crate::{
+    net::{default_backend::transport::TransportAddress, types::Role},
+    peer_manager::address_groups::get_address_group,
+    types::peer_id::PeerId,
+};
+
+use super::peer_context::PeerContext;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+struct NetGroupKeyed(u64);
+
+pub struct EvictionCandidate {
```

Comment:
Please add description/documentation for what's going on here and what this is used for and in what context.